src/zenml/stack_stores/sql_stack_store.py

```diff
@@ -245,6 +245,94 @@ def register_stack_component(
+
+        Raises:
+            KeyError: If no stack component exists with the given name.
+        """
+        with Session(self.engine) as session:
+            existing_components = session.exec(
+                select(ZenStackComponent).where(
+                    ZenStackComponent.component_type == component.type
+                )
+            ).all()
+            if not existing_components:
+                raise StackComponentExistsError(
+                    f"Unable to update stack component (type: {component.type}) "
+                    f"with name '{component.name}': No existing stack component "
+                    f"found with this name."
+                )
+            updated_component = session.exec(
+                select(ZenStackComponent)
+                .where(ZenStackComponent.component_type == component.type)
+                .where(ZenStackComponent.name == component.name)
+            ).first()
+            if updated_component:
+                updated_component.configuration = component.config
+
+            session.add(updated_component)
+            session.commit()
+            logger.info(
+                "Updated stack component with type '%s' and name '%s'.",
+                component.type,
+                component.name,
+            )
+
+    def rename_stack_component(
+        self,
+        old_name: str,
+        component: StackComponentWrapper,
+    ) -> None:
+        """Rename a stack component."""
+        with Session(self.engine) as session:
+            current_component = session.exec(
+                select(ZenStackComponent)
+                .where(ZenStackComponent.component_type == component.type)
+                .where(ZenStackComponent.name == old_name)
+            ).one_or_none()
+            if not current_component:
+                raise StackComponentExistsError(
+                    f"Unable to rename stack component (type: {component.type}) "
+                    f"with name '{component.name}': No existing stack component "
+                    f"found with this name."
+                )
+
+            # update the component itself
+            current_component.name = component.name
+            session.add(current_component)
+
+            # update the ZenStackDefinitions
+            stack_definitions = session.exec(
+                select(ZenStackDefinition)
+                .where(ZenStackDefinition.component_type == component.type)
+                .where(ZenStackDefinition.component_name == old_name)
+            ).all()
+            for definition in stack_definitions:
+                definition.component_name = component.name
+                session.add(definition)
+
+            session.commit()
+            logger.info(
+                "Renamed stack component with type '%s' and new name '%s'.",
+                component.type,
+                component.name,
+            )
+
+    def update_stacks_after_rename(
```

Comment:
This just goes to support what I was saying earlier about this method being an internal implementation detail that shouldn't be exposed through the base store API :) .

Reply:
Fixed.